packages/flutter/lib/src/rendering/auto_layout.dart

```diff
@@ -106,40 +118,65 @@ class AutoLayoutParentData extends ContainerBoxParentDataMixin<RenderBox> {
   /// may return null.
   List<al.Constraint> _constructImplicitConstraints() {
     return <al.Constraint>[
-      _rect._left >= al.cm(0.0), // The left edge must be positive.
-      _rect._right >= _rect._left, // Width must be positive.
+      _rect.left >= al.cm(0.0), // The left edge must be positive.
+      _rect.right >= _rect.left, // Width must be positive.
       // Why don't we need something similar for the top and the bottom?
     ];
   }
 }
 
+/// An abstract interface to control the layout of a [RenderAutoLayout].
```

Comment:
should have some comment about it being something you subclass, as opposed to a framework implementation detail